plugins/src/main/java/com/google/fhir/gateway/plugin/PatientAccessChecker.java

```diff
@@ -154,7 +154,9 @@ private AccessDecision checkPatientAccessInUpdate(RequestDetailsReader requestDe
   private AccessDecision processBundle(RequestDetailsReader requestDetails) {
     BundlePatients patientsInBundle = patientFinder.findPatientsInBundle(requestDetails);
 
-    if (patientsInBundle == null || patientsInBundle.areTherePatientToCreate()) {
+    if (patientsInBundle == null
+        || patientsInBundle.areTherePatientToCreate()
+        || !patientsInBundle.getDeletedPatients().isEmpty()) {
```

Comment:
So since there is no change to `ListAccessChecker` I am making this comment here: What happens if a patient is deleted for a `ListAccessChecker`? It seems that currently we let that go through (if the patient is in the access-list) but do not update the corresponding access `List`; should we update the list too? I mean should we change the [`postProcess`](https://github.com/google/fhir-gateway/blob/bff8dcc8df5c4913dd25754affa62a471d9c30b7/plugins/src/main/java/com/google/fhir/gateway/plugin/AccessGrantedAndUpdateList.java#L71) here such that the deleted patients are removed from the access list too? I am thinking that this same comment also applies to PR #71 but I forgot to mention it.

Reply:
Deletion of a patient will only soft delete that patient. The user won't be able to access the history for that resource. 
I didn't delete from the access list because of this reason. 
We should remove from the access list if we don't support fetching resource history. wdyt ?